hawc/apps/epiv2/models.py

```diff
@@ -0,0 +1,322 @@
+
+    name = models.CharField(
+        max_length=64,
+        help_text="A unique name for this adjustment factor that will help you identify it later.",
+    )
+    study_population = models.ForeignKey(
+        StudyPopulation, on_delete=models.CASCADE, related_name="adjustment_factors"
+    )
+    description = models.CharField(max_length=128, blank=True, null=True,)
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.name
+
+
+class DataExtraction(models.Model):
+    objects = managers.DataExtractionManager()
+
+    study_population = models.ForeignKey(
+        StudyPopulation, on_delete=models.CASCADE, related_name="data_extractions"
+    )
+    sub_population = models.CharField(max_length=64,)
+    outcome = models.ForeignKey(Outcome, on_delete=models.CASCADE, blank=True, null=True,)
+    exposure_level = models.ForeignKey(
+        ExposureLevel, on_delete=models.SET_NULL, blank=True, null=True
+    )
+    measurement_timing = models.CharField(max_length=128, blank=True, null=True,)
+    n = models.PositiveIntegerField(blank=True, null=True,)
+    effect_estimate_type = models.CharField(
+        max_length=128, choices=constants.EffectEstimateType.choices, blank=True, null=True,
+    )
+    effect_description = models.CharField(
+        max_length=128,
+        blank=True,
+        null=True,
+        verbose_name="Effect estimate description",
+        help_text="Description of the effect estimate with units, including comparison group if applicable",
+    )
+    effect_estimate = models.FloatField(blank=True, null=True,)
+    exposure_rank = models.PositiveSmallIntegerField(
+        blank=True,
+        null=True,
+        help_text="Rank this comparison group by exposure (lowest exposure group = 1)",
+    )
+    ci_lcl = models.FloatField(verbose_name="Confidence Interval LCL", blank=True, null=True,)
+    ci_ucl = models.FloatField(verbose_name="Confidence Interval UCL", blank=True, null=True,)
+    sd_or_se = models.FloatField(
+        verbose_name="Standard Deviation or Standard Error", blank=True, null=True,
+    )
+    pvalue = models.CharField(verbose_name="p-value", max_length=128, blank=True, null=True,)
+    significant = models.BooleanField(
+        verbose_name="Statistically Significant",
+        choices=constants.SIGNIFICANT_CHOICES,
+        blank=True,
+        null=True,
+    )
+    adjustment_factor = models.ForeignKey(
+        AdjustmentFactor, on_delete=models.SET_NULL, blank=True, null=True,
+    )
+    confidence = models.CharField(
+        max_length=128, verbose_name="Study confidence", blank=True, null=True,
+    )
+    data_location = models.CharField(
+        max_length=128, help_text="e.g., table number", blank=True, null=True,
+    )
+    statistical_method = models.CharField(max_length=128, blank=True, null=True,)
+    comments = models.CharField(max_length=128, blank=True, null=True,)
```

Comment:
make this a TextField; no length required; do the same for others where the text length may be hard to pin-down